opte/src/engine/packet.rs

```diff
@@ -1865,7 +1887,23 @@ impl PacketSeg {
     }
 
     fn link(&mut self, seg: &PacketSeg) {
-        unsafe { (*self.mp).b_cont = seg.mp };
+        unsafe {
+            // We should not be creating message block continuations to zero
+            // sized blocks. This is not a generally expected thing and has
+            // caused NIC hardware to stop working. Stopping short of a
+            // production panic, but this should fail any tests.
+            #[cfg(any(feature = "std", test))]
```

Comment:
Maybe change this to a `debug_assert_ne!`?

Reply:
Yup, that's definitely nicer. Done.